### ISSUE TITLE
Pull empty repository

### ISSUE THREAD

**Issue by larshp** [larshp/abapGit#444]
Steps to reproduce:
clone https://github.com/larshp/Empty.git

After cloning the "Pull" link appears:

![image](https://cloud.githubusercontent.com/assets/5888506/20463758/e181fc96-af3a-11e6-9449-0fbd391bc676.png)

Click "Pull"

Expected result: "Pull" link not shown

**Comment by sbcgua** [comment]
A bit tricky issue. That's because of .abapgit. It has no representation locally - generated each time, and has no item. So it is not stored to state. So has MM state (modified from both sides) and triggers Pull.
I think we should should just add .abapgit to state as an exception. At the end it is code-relevant in some way. Does it seems logical ? (can fix if yes)

**Comment by larshp** [comment]
I havent looked into the details, but adding .abapgit.xml to state sounds ok

TL;DR: yes

**Comment by sbcgua** [comment]
fixed .abapgit in local state.  But ! The issue didn't go completely. 
Now because of foobar.txt - it is not ignored. And cannot be deserialized. So it always appear as a new file. But this time I think it is OK and not a bug. So if OK, I'll PR and close.

**Comment by larshp** [comment]
ok for now

will this also fix https://github.com/larshp/abapGit/issues/440 ?

**Comment by sbcgua** [comment]
440: on the first glance - no. But I'm not sure it related to caching, I didn't look much into .abapgit logic also.

**Comment by larshp** [comment]
retested, problem still exists,

![image](https://cloud.githubusercontent.com/assets/5888506/22403427/07fdb58c-e617-11e6-923c-139906e33a1d.png)

**Comment by larshp** [comment]
tested ok, guess this was fixed a while ago with the local .abapgit.xml state

![image](https://cloud.githubusercontent.com/assets/5888506/26774374/46429b7e-49d0-11e7-9c26-e32ccee6020b.png)
